app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkChaptersCompletedFragmentTest.kt

```diff
@@ -0,0 +1,894 @@
+      scrollToPosition(position = 1)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 1)
+      scrollToPosition(position = 2)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.mark_chapters_completed_recycler_view,
+          position = 2,
+          targetViewId = R.id.mark_chapters_completed_chapter_check_box
+        )
+      ).check(matches(not(isEnabled())))
+    }
+  }
+
+  @Test
+  fun testMarkChaptersCompletedFragment_configChange_firstStory_selectChaptersInOrder_unselectFirstChapter_secondChapterIsDisabled() { // ktlint-disable max-line-length
+    launch<MarkChaptersCompletedTestActivity>(
+      createMarkChaptersCompletedTestActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
+      scrollToPosition(position = 1)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 1)
+      scrollToPosition(position = 2)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 2)
+      scrollToPosition(position = 1)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 1)
+      scrollToPosition(position = 2)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.mark_chapters_completed_recycler_view,
+          position = 2,
+          targetViewId = R.id.mark_chapters_completed_chapter_check_box
+        )
+      ).check(matches(not(isEnabled())))
+    }
+  }
+
+  @Test
+  fun testMarkChaptersCompletedFragment_selectAllChapters_allCheckBoxIsChecked() {
+    launch<MarkChaptersCompletedTestActivity>(
+      createMarkChaptersCompletedTestActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 1)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 1)
+      scrollToPosition(position = 2)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 2)
+      scrollToPosition(position = 4)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 4)
+      scrollToPosition(position = 6)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 6)
+      scrollToPosition(position = 7)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 7)
+      scrollToPosition(position = 9)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 9)
+      scrollToPosition(position = 10)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 10)
+      scrollToPosition(position = 12)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 12)
+      scrollToPosition(position = 13)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 13)
+      onView(withId(R.id.mark_chapters_completed_all_check_box)).check(matches(isChecked()))
+    }
+  }
+
+  @Test
+  fun testMarkChaptersCompletedFragment_configChange_selectAllChapters_allCheckBoxIsChecked() {
```

Comment:
Ditto as above. On a counter thought, are we doing this to check if it works in landscape as well? If yes, can we include tests to check if the outcome persists on configChange?

Reply:
No, we are checking if the outcomes persist on configChange. I have updated the test name now. PTAL. Thanks.